src/libcommonserver/log/customrollingfileappender.cpp

```diff
@@ -286,30 +288,45 @@ void CustomRollingFileAppender::rollover(bool alreadyLocked) {
 }
 
 void CustomRollingFileAppender::checkForExpiredFiles() {
-    // Delete expired files
-    if (_expire > 0) {
-        _lastExpireCheck = std::chrono::system_clock::now();
-        QDateTime now = QDateTime::currentDateTime();
-
-        IoError ioError = IoErrorSuccess;
-        SyncPath logDirPath;
-        if (!IoHelper::logDirectoryPath(logDirPath, ioError)) {
-            throw std::runtime_error("Error in CustomRollingFileAppender: failed to get the log directory path.");
+    _lastExpireCheck = std::chrono::system_clock::now();
+    // Archive Previous Log Files and delete expired files
+    IoError ioError = IoErrorSuccess;
+    SyncPath logDirPath;
+    if (!IoHelper::logDirectoryPath(logDirPath, ioError) || ioError != IoErrorSuccess) {
+        throw std::runtime_error("Error in CustomRollingFileAppender: failed to get the log directory path.");
+    }
+    IoHelper::DirectoryIterator dirIt(logDirPath, false, ioError);
+    bool endOfDir = false;
+    DirectoryEntry entry;
+    while (dirIt.next(entry, endOfDir, ioError) && !endOfDir && ioError == IoErrorSuccess) {
+        FileStat fileStat;
+        IoHelper::getFileStat(entry.path(), &fileStat, ioError);
+        if (ioError != IoErrorSuccess || fileStat.nodeType != NodeTypeFile) {
+            continue;
         }
 
-        QDir logDir(SyncName2QStr(logDirPath.native()));
-        QStringList filters;
-        filters << SyncName2QStr(Str("*") + Utility::logFileName() + Str("*"));
-        QStringList logFileNameList = logDir.entryList(filters, QDir::Files);
-        for (const QString &logFileName : logFileNameList) {
-            QFileInfo fileInfo(logDir, logFileName);
-            QDateTime expireDateTime = fileInfo.lastModified().addSecs(_expire * 3600);
+        // Delete expired files
+        if (_expire > 0 && entry.path().string().find(APPLICATION_NAME) != std::string::npos) {
+            auto now = std::chrono::system_clock::now();
```

Comment:
```suggestion
            const auto now = std::chrono::system_clock::now();
```
Same comment for the next two variables.